### ISSUE TITLE
Nuclei stops working.

### ISSUE THREAD

**Issue by efrann** [projectdiscovery/nuclei#5204]
### Nuclei version: 
3.2.7

### Current Behavior: 
When I started a scan, I thought that it was stopping suddenly cause I wasn't find anything and I add -stats parameter to check it. This time when it was 6% the scan is stopped suddenly.  As you can see on the screenshot there is a bunch of request needed to be sent. 68362/1122527

### Expected Behavior: 
Complete the scan 

### Steps To Reproduce: 
nuclei -l input_file -s medium,high,critical -et nuclei-templates/dns -t nuclei-templates  -stats -v | tee nuclei_2.txt

![Screenshot 2024-05-20 at 23 46 59](https://github.com/projectdiscovery/nuclei/assets/44778301/c5cefbc3-c424-406e-aa65-8d456ddbfa47)

### Anything else:

The input file has just 118 url. I am using macbook m1.

**Comment by efrann** [comment]
@jdewald thanks for your response , this time I stucked 75%. but I didn't get that INF info messages. I only get
`[INF] Dumped HTTP request for <>url`

I realized that after getting tons of
`[WRN] Could not make network request for (<url>) : could not connect to server: [:RUNTIME] ztls fallback failed <- dial tcp <IP>: i/o timeout`
warnings, My scan is stopping. But I am not sure how to solve this problem.
